xaitk_saliency/utils/bin/sal_on_coco_dets.py

```diff
@@ -69,21 +78,17 @@ def det_sal_on_img_dir(
 
         exit()
 
-    files = os.listdir(img_dir)
-
-    img_files = []
-    imgs = []
+    if not is_usable:
+        print("This tool requires additional dependencies, please install 'xaitk-saliency[tools]'")
+        exit()
```

Comment:
This should likely not exit with a 0 return code as this is not a successful condition.